extensions/github/src/create-issue-branch.tsx

```diff
@@ -0,0 +1,143 @@
+        const createResult = await github.createLinkedBranch({
+          input: values.name === "" ? payload : { ...payload, name: values.name },
+        });
+        const branchName = createResult.createLinkedBranch?.linkedBranch?.ref?.name;
+
+        toast.style = Toast.Style.Success;
+        toast.title = `${branchName} is created`;
+
+        if (branchName) {
+          toast.primaryAction = {
+            title: "Copy Branch Name",
+            shortcut: { modifiers: ["shift", "cmd"], key: "c" },
+            onAction: () => Clipboard.copy(branchName),
+          };
+          toast.secondaryAction = {
+            title: "Open in Browser",
+            shortcut: { modifiers: ["shift", "cmd"], key: "o" },
+            onAction: async () => await open(`${repositoryUrl}/tree/${branchName}`),
+          };
+        }
+      } catch (error) {
+        toast.style = Toast.Style.Failure;
+        toast.title = "Failed creating a branch";
+        toast.message = getErrorMessage(error);
+      }
+    },
+    initialValues: {
+      repository: draftValues?.repository ?? "",
+      issue: draftValues?.issue ?? "",
+      name: draftValues?.name || "",
+      oid: draftValues?.oid ?? "",
+    },
+    validation: {
+      repository: FormValidation.Required,
+      issue: FormValidation.Required,
+    },
+  });
+
+  const { data, isLoading: repositoryIsLoading } = useCachedPromise(
+    (repository) => {
+      const selectedRepository = repositories?.find((r) => r.id === repository);
+
+      if (!selectedRepository) {
+        return Promise.resolve(null);
+      }
+
+      return github.repositoryIssues({ owner: selectedRepository.owner.login, name: selectedRepository.name });
+    },
+    [values.repository],
+    { execute: !!values.repository }
+  );
+
+  const issues = data?.repository?.issues?.nodes?.filter((node) => node?.linkedBranches.totalCount == 0);
+
+  useEffect(() => {
+    data?.repository?.defaultBranchRef?.target?.oid && setOid(data.repository.defaultBranchRef.target.oid);
+    data?.repository?.url && setRepositoryUrl(data.repository.url);
+  }, [data]);
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            onSubmit={handleSubmit}
+            title="Create Branch"
+            icon={{ source: "branch.svg", tintColor: Color.PrimaryText }}
+          />
+        </ActionPanel>
+      }
+      enableDrafts
+    >
+      <Form.Dropdown {...itemProps.repository} title="Repository" isLoading={repositoriesIsLoading} storeValue>
+        {repositories?.map((repository) => (
+          <Form.Dropdown.Item
+            key={repository.id}
+            title={repository.nameWithOwner}
+            value={repository.id}
+            icon={{ source: repository.owner.avatarUrl, mask: Image.Mask.Circle }}
+          />
+        ))}
+      </Form.Dropdown>
+
+      <Form.Separator />
+
+      <Form.Dropdown {...itemProps.issue} title="Issue" isLoading={repositoryIsLoading}>
```

Comment:
Maybe `Linked Issue` would be more explicit?

Reply:
It's an issue field that's going to be linked once the branch is created, so changing label to Linked Issue in the form would be little confusing?